### PR TITLE
docs: improve gRPC-JSON filter doc

### PR DESCRIPTION
**Pull request by ctoomey**
Signed-off-by: ctoomey <ctoomey@livongo.com>

For an explanation of how to fill out the fields, please see the relevant section 
in PULL_REQUESTS.md

*Description*: improve gRPC-JSON filter doc
*Risk Level*: Low
*Testing*: Viewed doc on github
*Docs Changes*:  Added link to protobuf JSON mapping, added doc for how stream request/response params handled, added sample config
*Release Notes*:
[Optional Fixes #Issue]
[Optional *Deprecated*:]

### CODE REVIEW

File: docs/root/configuration/http_filters/grpc_json_transcoder_filter.rst [hunk 1]
```diff
@@ -74,4 +81,81 @@ as its output message type. The implementation needs to set
...
+.. code-block:: yaml
+
+  admin:
+    access_log_path: /tmp/admin_access.log
+    address:
+      socket_address: { address: 0.0.0.0, port_value: 9901 }
+
+  static_resources:
+    listeners:
+    - name: listener1
```
Comment (lizan) [inline docs/root/configuration/http_filters/grpc_json_transcoder_filter.rst]: I don't think you need this, listener2 can serve both gRPC and its RESTful JSON mapping.

File: docs/root/configuration/http_filters/grpc_json_transcoder_filter.rst [hunk 2]
```diff
@@ -74,4 +81,62 @@ as its output message type. The implementation needs to set
...
+              virtual_hosts:
+              - name: local_service
+                domains: ["*"]
+                routes:
+                - match: { prefix: "/" }
+                  route: { cluster: grpc, timeout: { seconds: 60 } }
+            http_filters:
+            - name: envoy.grpc_json_transcoder
+              config:
+                  proto_descriptor: "/tmp/envoy/proto.pb"
```
Comment (lizan) [inline docs/root/configuration/http_filters/grpc_json_transcoder_filter.rst]: nit: 2 space indent

File: docs/root/configuration/http_filters/grpc_json_transcoder_filter.rst [hunk 3]
```diff
@@ -74,4 +81,62 @@ as its output message type. The implementation needs to set
...
+                  services: ["HelloWorld"]
+                  print_options:
+                      add_whitespace: true
```
Comment (lizan) [inline docs/root/configuration/http_filters/grpc_json_transcoder_filter.rst]: ditto

**Comment by ctoomey** [comment]
Fixed config spacing

**Pull request merged**